src/thunderbots/software/grsim_communication/visitor/grsim_command_primitive_visitor.cpp

```diff
@@ -29,7 +29,35 @@ void GrsimCommandPrimitiveVisitor::visit(const ChipPrimitive &chip_primitive)
 void GrsimCommandPrimitiveVisitor::visit(
     const DirectVelocityPrimitive &direct_velocity_primitive)
 {
-    // TODO: https://github.com/UBC-Thunderbots/Software/issues/97
+    // get current robot position and orientation(angle)
+    Point robot_position    = robot.position();
+    Angle robot_orientation = robot.orientation();
+
+    // create linear velocity vector from direct velocity primitive
+    Vector linear_velocity_in_robot_coordinates =
+        Vector(direct_velocity_primitive.getXVelocity(),
+               direct_velocity_primitive.getXVelocity());
+
+    // transfer velocity into global coordinate by rotating the vector in robot
+    // coordinates by the angle of robot
+    Vector linear_velocity_in_global_coordinates =
+        linear_velocity_in_robot_coordinates.rotate(robot.orientation());
+
+    // TODO find reasonable scalars to multiply linear and angular velocity to get
+    // reasonable speed
```

Comment:
Could we either resolve this TODO or assign an issue to it?

Reply:
Currently I'm not sure how to get the reasonable scalar.

Reply:
@MathewMacDougall?

Reply:
I just made a new ticket. Link this TODO to #286.

@garethellis0 The problem is essentially that we want to directly set/request the robot have a certain velocity, and can't do that with the current motion controller interface. It's very hard/impossible to set the requested "final state" of the robot such that the robot ends up with the desired velocities.

Reply:
Ah, unfortunate. How are we testing that this is working correctly if the motion controller interface isn't compatible with it then?

Reply:
Visual inspection in grsim that it's close enough for now.

Reply:
:+1: